cdc server panic: invalid memory address or nil pointer dereference

### What did you do?

run cdc server 

### What did you expect to see?

cdc server start normally

### What did you see instead?

```log
[WARN] TiCDC server data-dir is not set. Please use `cdc server --data-dir` to start the cdc server if possible.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x2e96b49]

goroutine 1 [running]:
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).resign(0x41ae0c0?, {0x41ae0f8?, 0xc0007d1680?})
        github.com/pingcap/tiflow/cdc/capture/capture.go:577 +0x29
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).AsyncClose(0xc000b37320)
        github.com/pingcap/tiflow/cdc/capture/capture.go:618 +0x48d
github.com/pingcap/tiflow/cdc/server.(*server).Close(0xc0009aec80)
        github.com/pingcap/tiflow/cdc/server/server.go:407 +0x50
github.com/pingcap/tiflow/pkg/cmd/server.(*options).run(0xc000147570, 0xc0007c6300?)
        github.com/pingcap/tiflow/pkg/cmd/server/server.go:163 +0x6bc
github.com/pingcap/tiflow/pkg/cmd/server.NewCmdServer.func1(0xc0007c6300?, {0xc0007d0240?, 0x6?, 0x6?})
        github.com/pingcap/tiflow/pkg/cmd/server/server.go:306 +0x59
github.com/spf13/cobra.(*Command).execute(0xc0007c6300, {0xc0007d0180, 0x6, 0x6})
        github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc0007c6000)
        github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.6.1/command.go:968
github.com/pingcap/tiflow/pkg/cmd.Run()
        github.com/pingcap/tiflow/pkg/cmd/cmd.go:50 +0x17e
main.main()
        ./main.go:22 +0x17
```

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
Release Version: v6.6.0-alpha
Git Commit Hash: 099da1e08f1310c2366093f2c314a51e0cde772b
Git Branch: heads/refs/tags/v6.6.0-alpha
UTC Build Time: 2022-12-14 11:43:03
Go Version: go version go1.19.3 linux/amd64
Failpoint Build: false

```

--- a/cdc/capture/capture.go
+++ b/cdc/capture/capture.go
@@ -209,6 +209,11 @@ func (c *captureImpl) reset(ctx context.Context) error {
 
 	c.captureMu.Lock()
 	defer c.captureMu.Unlock()
+	c.info = &model.CaptureInfo{
+		ID:            uuid.New().String(),
+		AdvertiseAddr: c.config.AdvertiseAddr,
+		Version:       version.ReleaseVersion,
+	}
 	c.EtcdClient = etcdClient
 	c.migrator = migrate.NewMigrator(c.EtcdClient, c.pdEndpoints, c.config)
 
@@ -222,12 +227,6 @@ func (c *captureImpl) reset(ctx context.Context) error {
 		concurrency.WithLease(lease.ID))
 	if err != nil {
 		return cerror.WrapError(cerror.ErrNewCaptureFailed, err)
-	}
-
-	c.info = &model.CaptureInfo{
-		ID:            uuid.New().String(),
-		AdvertiseAddr: c.config.AdvertiseAddr,
-		Version:       version.ReleaseVersion,
 	}
 
 	if c.upstreamManager != nil {

--- a/cdc/capture/capture.go
+++ b/cdc/capture/capture.go
@@ -574,6 +574,9 @@ func (c *captureImpl) resign(ctx context.Context) error {
 	failpoint.Inject("capture-resign-failed", func() {
 		failpoint.Return(errors.New("capture resign failed"))
 	})
+	if c.election == nil {
+		return nil
+	}
 	return cerror.WrapError(cerror.ErrCaptureResignOwner, c.election.resign(ctx))
 }
 

